extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -188,6 +202,7 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    this.lifecycleReporter_.sendPing('layoutCallback');
```

Comment:
Yes, we're definitely doing different things in them.  That's precisely why we need this.

First, note that, while we're returning timings, at this point we're _much_ more interested in the count of pings at each point in the lifecycle of each path.

The issue is that we're currently seeing experimental results that we can't disambiguate -- they might indicate a problem in A4A or they might indicate everything working as intended.  We can't tell because we can't currently figure out how many ad slots there are, how many ads are requested, and how many are rendered in each path (3p vs A4A).  We can't tell if some things are getting lost b/c of an unknown bug in the middle of the A4A promise chain, or if all ad requests are getting through okay but the counts are weird because of server-side issues.  We're trying to come up with some base statistics that will tell us at least that.

Counting the number of `onLayoutMeasure` will tell us how many ad slots there are in each path, so tell us whether we have baseline comparability.  That also tells us how many ad requests *should* be made by A4A, which we can match against `buildUrl` and `sendXHRRequest` to be sure that the ad requests are actually going out.

In the 3p path, `layoutCallback` is what writes the 3p iframe, thus triggering the ad request and render.  That's the closest proxy we have for 'number of 3p ads that get requested/rendered', taking into account the 3 viewport limit, the chance of the user swiping away before layout, etc.  We want to compare that to the number of renders according to A4A's `maybeRenderAmpAd_` plus `renderViaIframe`.